samples/nrf9160/mqtt_simple/src/main.c

```diff
@@ -444,26 +480,36 @@ static void modem_configure(void)
 		/* Wait for the LWM2M_CARRIER to configure the modem and
 		 * start the connection.
 		 */
-		printk("Waitng for carrier registration...\n");
+		LOG_INF("Waitng for carrier registration...");
 		k_sem_take(&carrier_registered, K_FOREVER);
-		printk("Registered!\n");
+		LOG_INF("Registered!");
 #else /* defined(CONFIG_LWM2M_CARRIER) */
 		int err;
 
-		printk("LTE Link Connecting ...\n");
+		LOG_INF("LTE Link Connecting...");
 		err = lte_lc_init_and_connect();
-		__ASSERT(err == 0, "LTE link could not be established.");
-		printk("LTE Link Connected!\n");
+		if (err) {
+			LOG_INF("Failed to establish LTE connection: %d", err);
+			return err;
+		}
+		LOG_INF("LTE Link Connected!");
 #endif /* defined(CONFIG_LWM2M_CARRIER) */
 	}
 #endif /* defined(CONFIG_LTE_LINK_CONTROL) */
+
+	/* Turn off power saving modes for a more responsive demo */
```

Comment:
Without this, messages sent to the device are not received until `poll` times out.
Also, when running with the LWM2M carrier, an MQTT disconnect always occurs.  I assume this is due to some latency and the MQTT connection is not pinged on time (> keep alive value).

Is there a better way to handle this?

Reply:
> Without this, messages sent to the device are not received until poll times out.

@jtguggedal Do you know if this is some known issue?

Reply:
The reason is that with PSM, the device is not reachable before the PWM interval has passed, or alternatively when the device initiates data transfer, which is what happens on `poll` timeout when `PINGREQ` or now also if the button is pressed. 
If the intention is a responsive demo, I agree that it makes sense to disable both PSM and eDRX. Could add a note about in the docs and possibly a print in the logs.

Reply:
update readme and added log.